strimzi/04-service-binding.yaml

```
@@ -0,0 +1,81 @@
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: kafka-producer-binding-user
+spec:
+  application:
+    apiVersion: app.stacks/v1beta1
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: kafka.strimzi.io/v1beta1
+    kind: KafkaUser
+    name: my-producer-user
+---
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: kafka-producer-binding-cluster
+spec:
+  application:
+    apiVersion: app.stacks/v1beta1
+    kind: RuntimeComponent
+    name: kafka-producer
+  service:
+    apiVersion: kafka.strimzi.io/v1beta1
+    kind: Kafka
+    name: my-producer-cluster
+    mappings:
+      - name: hosts
+        value: |-
+          {{- range .status.listeners -}}
+            {{- if and (eq .type "tls") (gt (len .addresses) 0) -}}
+              {{- with (index .addresses 0) -}}
+                {{ .host }}:{{ .port }}
+              {{- end -}}
+            {{- end -}}
+          {{- end -}}
+    env:
+      - name: BOOTSTRAP_SERVERS
+        key:  hosts      
+---
+  apiVersion: service.binding/v1alpha1
+  kind: ServiceBinding
+  metadata:
+    name: kafka-consumer-binding-user
+  spec:
+    application:
+      apiVersion: app.stacks/v1beta1
+      kind: RuntimeComponent
+      name: kafka-consumer
+    service:
+      apiVersion: v1beta1/kafka.strimzi.io
+      kind: KafkaUser
+      name: my-consumer-user
+  ---
+  apiVersion: service.binding/v1alpha1
+  kind: ServiceBinding
+  metadata:
+    name: kafka-consumer-binding-cluster
+  spec:
+    application:
+      apiVersion: app.stacks/v1beta1
+      kind: RuntimeComponent
+      name: kafka-consumer
+    service:
+      apiVersion: kafka.strimzi.io/v1beta1
+      kind: Kafka
+      name: my-consumer-cluster
```

I think both the producer and the consumer will be exchanging messages with the same Kafka Cluster, `my-cluster`:
```suggestion
    name: my-cluster
```

---

good catch!